### PR TITLE
Reuse cluster

### PR DESCRIPTION
**Pull request by larisau**
Run SCT on existing cluster

[addition by Roy]:
This PR should add SCT the ability to re-run a test on an existing cluster and do not provision a new cluster during the setup phase.

It should help us in many cases where we run longevity for few days and something happened to the test (Jenkins crushed, unhandled exception in one of the nemesis, etc..).

The way we want to use it, is providing the list of cluster IPs in the test yaml and let the SCT load the cluster objects and run the test again (e.g. run the load and start nemesis for longevity).

### CODE REVIEW

File: sdcm/cluster_gce.py [hunk 1]
```diff
@@ -197,64 +201,90 @@ def _get_persistent_disk_struct(self, name, disk_size, disk_type='pd-ssd', dc_id
...
-                self.nodes += [n]
-
-                local_nodes = [n for n in self.nodes if n.dc_idx == dc_idx]
-                if len(local_nodes) > len(nodes):
-                    n.is_addition = True
-
-        assert len(nodes) == count, 'Fail to create {} instances'.format(count)
-        self.log.info('added nodes: %s', nodes)
+        if cluster.Setup.REUSE_CLUSTER:
+            instances = self._get_instances()
```
Comment (bentsi) [inline sdcm/cluster_gce.py]: I think cluster Base class should have _instances_ property when the instances are stored

File: sdcm/cluster.py [hunk 2]
```diff
@@ -60,8 +60,6 @@
 EC2_INSTANCES = []
```
Comment (bentsi) [inline sdcm/cluster.py]: Actually we don't need this too...

File: sdcm/tester.py [hunk 3]
```diff
@@ -289,6 +289,10 @@ def setUp(self):
                                              'loaders': [getattr(n, ip_addr_attr) for n in self.loaders.nodes]},
                                     scylla_version=self.db_cluster.nodes[0].scylla_version,
                                     **mgmt_params)
+
```
[the remark this answers is not in the document]
Comment (larisau) [reply sdcm/tester.py]: agree

[the remark this answers is not in the document]
Comment (larisau) [reply sdcm/tester.py]: added

File: sdcm/cluster.py [hunk 4]
```diff
@@ -1807,6 +1807,7 @@ def node_setup(self, node, verbose=False, db_node_address=None):
         node.wait_ssh_up(verbose=verbose)
 
         if Setup.REUSE_CLUSTER:
+            self.kill_stress_thread()
```
Comment (bentsi) [inline sdcm/cluster.py]: :+1:

**Comment by roydahan** [comment]
@larisau, please send email to our group describing the new feature and how to use it.
thx.

**Pull request merged**